src/local/room_position.rs

```diff
@@ -0,0 +1,375 @@
+//! Room position type and related operations and traits.
+//!
+//! This is a reimplementation/translation of the `RoomPosition` code originally
+//! written in JavaScript. All RoomPosition to RoomPosition operations in this
+//! file stay within Rust.
+use std::fmt;
+
+use super::LocalRoomName;
+
+mod extra_math;
+mod game_math;
+mod pair_utils;
+mod world_utils;
+
+const HALF_WORLD_SIZE: i32 = 128;
```

Comment:
What is this meant to represent? I thought it was to wrap around from `i8` to `u8`.

Reply:
It's a magic number copied from the JavaScript implementation, mostly. I've added a comment.